code/game/objects/items/devices/flashlight.dm

```diff
@@ -206,3 +229,60 @@
 
 /obj/item/device/flashlight/slime/attack_self(mob/user)
 	return //Bio-luminescence does not toggle.
+
+//Glowsticks
+
+/obj/item/device/flashlight/flare/glowstick
+	name = "green glowstick"
+	desc = "A green military-grade glowstick."
+	w_class = 2
+	brightness_on = 4
+	light_power = 2
+	light_color = "#49F37C"
+	icon = 'icons/obj/glowsticks.dmi'
+	icon_state = "glowstick"
+	item_state = "glowstick"
+	on_damage = 0
+	produce_heat = 0
+	contained_sprite = 1
+
+/obj/item/device/flashlight/flare/glowstick/attack_self(mob/user)
+
+	if(!fuel)
+		user << "<span class='notice'>The glowstick has already been turned on.</span>"

```

Comment:
This is a relatively odd feedback message. `!fuel` means it's empty and no longer glowing, right? At which point, it'd be more clear if it said, `\The [src] has already been used.` or expended, or something like that.